File: Blind 75/Arrays/Contains Duplicate.java
class Solution {
    public boolean containsDuplicate(int[] nums) {
        // Time: O(n) 
        // Because it iterates through the array once

        // Space: O(n)
        // Because the space used by a hash set is linear
        // with the nubmer of elements in nums

        HashSet<Integer> setOfNums = new HashSet<>();
        for (int num : nums) {
            if (setOfNums.contains(num)) return true;
            setOfNums.add(num);
        }
        return false;
    }
}

File: Python Blind 75/Arrays/Contains Duplicate.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        # Time: O(n), Space: O(n)
        seenNums = set()

        for num in nums:
            if num in seenNums:
                return True
            else:
                seenNums.add(num)
        
        return False
